hack/verify-protobuf.sh

```diff
@@ -13,7 +13,7 @@ cleanup
 
 for gv in ${API_GROUP_VERSIONS}; do
   mkdir -p "${TMP_ROOT}/${gv}"
-  cp -a --no-preserve=timestamp "${SCRIPT_ROOT}/${gv}"/* "${TMP_ROOT}/${gv}"
+  cp -a "${SCRIPT_ROOT}/${gv}"/* "${TMP_ROOT}/${gv}"
```

Comment:
Why is this commit modifying `hack/verify-protobuf.sh`?  It isn't clear from the commit message.

Reply:
Mac OS X does not provide support for --no-preserve=timestamp so had to remove it.
For example:

 cp -a --no-preserve=timestamp "${SCRIPT_ROOT}/${gv}"/* "${TMP_ROOT}/${gv}"
cp: illegal option -- -
usage: cp [-R [-H | -L | -P]] [-fi | -n] [-aclpSsvXx] source_file target_file
       cp [-R [-H | -L | -P]] [-fi | -n] [-aclpSsvXx] source_file ... target_directory

However, you are right may be I should add a different commit for this. Shall I send a seperate PR for this.  ?

Reply:
We can't fix the Git history now that the PR has been merged, but in the future, it would be appropriate to use a separate PR, or at least a separate commit that explains why you are making the change.  A person looking through the Git history might think the the change resulted from a mistake during rebase or inadvertent change.  At least now there's a chance that that person will check the PR and see this discussion.  

I'm surprised this change wasn't flagged during review; maybe it was overlooked.  

@JoelSpeed, is removing `--no-preserve=timestamp` safe to do?  It had been added here: https://github.com/openshift/api/pull/719/commits/a2550b3dde9abcd09ef1336c6b284786e472322b

Reply:
I have previously tested this (I run a mac, tested with fedora container for comparison) and couldn't identify any negative impact of the flag.
I've also been working on moving from gengo v1 to v2. Once that is done (next week?), then I'm unblocked on my move of the proto generation stuff into the tools submodule to align with the rest of the tooling. That PR has been blocked for a while because of empty `GOPATH` support. Gengo v2 should have fixed this. 

Once that PR is in, the script will be rewritten completely anyway.

I agree it was unfortunate that it was in the same commit, I overlooked that part

Reply:
@JoelSpeed has the PR been in ? The reason I am asking is because I need to vendor the api in the cluster-ingress-operator as well. Would you please let me know once that PR is in ? cc @Miciah

Reply:
@miheer Not yet, are you blocked? What exactly are you blocked on?

Reply:
@JoelSpeed I just wanted to make sure that I add the API change in the ingress operator with the original hack scripts. So, once your PR is added which has my changes mentioned here plus the original hack script I can then `go mod edit`, `go mod tidy` and `go mod vendor` the cluster-ingress-operator with it.

Reply:
@miheer, we don't vendor `hack/verify-protobuf.sh`.  It is only used by CI for the openshift/api repository itself.  There's no need to wait for Joel's changes before proceeding with the vendor bump in openshift/cluster-ingress-operator.